src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -131,21 +129,29 @@ public IEnumerable<ITagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSp
                     var dataTagSpans = tag.Span.GetSpans(snapshot);
                     if (dataTagSpans.Count == 1)
                     {
-                        var dataTagSpan = dataTagSpans[0];
-                        var parameterHintUITag = InlineHintsTag.Create(
-                            tag.Tag.Hint, Format, _textView, dataTagSpan, _taggerProvider, _formatMap, classify);
-
-                        _cache.Add(new TagSpan<IntraTextAdornmentTag>(dataTagSpan, parameterHintUITag));
+                        _cache.Add((tag, null));
                     }
                 }
             }
 
+            var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var classify = document?.Project.Solution.Workspace.Options.GetOption(InlineHintsOptions.ColorHints, document?.Project.Language) ?? false;
+
             var selectedSpans = new List<ITagSpan<IntraTextAdornmentTag>>();
-            foreach (var tagSpan in _cache)
+            for (var i = 0; i < _cache.Count; i++)
             {
-                if (spans.IntersectsWith(tagSpan.Span))
+                var tagSpan = _cache[i].mappingTagSpan.Span.GetSpans(snapshot)[0];
+                if (spans.IntersectsWith(tagSpan))
                 {
-                    selectedSpans.Add(tagSpan);
+                    if (_cache[i].tagSpan is not { } hintTagSpan)
```

Comment:
```suggestion
                    if (_cache[i].tagSpan is null)
```

Reply:
changing to deconstructed foreach will change this slightly, but there's name collisions so it will depend on what you name the deconstructed parts

Reply:
This doesn't work for lazy initialization.